browser/playlist/test/playlist_browsertest.cc

```diff
@@ -333,3 +342,73 @@ IN_PROC_BROWSER_TEST_F(PlaylistBrowserTest, PlaylistTabHelper) {
   WaitUntil(base::BindLambdaForTesting(
       [&]() { return playlist_tab_helper->found_items().size() == 0; }));
 }
+
+class PlaylistBrowserTestWithArbitrarySites : public PlaylistBrowserTest {
+ public:
+  // PlaylistBrowserTest:
+  void SetUpHTTPSServer() override {
+    https_server_ = std::make_unique<net::EmbeddedTestServer>(
+        net::test_server::EmbeddedTestServer::TYPE_HTTPS);
+    https_server_->RegisterRequestHandler(base::BindRepeating(
+        &PlaylistBrowserTestWithArbitrarySites::Serve, https_server_.get()));
+    ASSERT_TRUE(https_server_->Start());
+  }
+
+ private:
+  static std::unique_ptr<net::test_server::HttpResponse> Serve(
+      net::EmbeddedTestServer* server,
+      const net::test_server::HttpRequest& request) {
+    GURL absolute_url = server->GetURL(request.relative_url);
+    auto response = std::make_unique<net::test_server::BasicHttpResponse>();
+    response->set_code(net::HTTP_OK);
+    response->set_content(R"html(
+        <html>
+        <meta property="og:image" content="/img.jpg">
+        <body>
+          <video src="test1.mp4"/>
+        </body>
+        </html>
+      )html");
+    response->set_content_type("text/html; charset=utf-8");
+    return response;
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(PlaylistBrowserTestWithArbitrarySites,
```

Comment:
I hope this test can describe well enough what happens when media detection is deferred,